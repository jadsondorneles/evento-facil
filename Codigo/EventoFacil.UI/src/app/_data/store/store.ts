import { create } from 'zustand'
import { devtools } from 'zustand/middleware'

import { SnackbarSlice, snackbarSlice } from './slices/snackbarSlice'

type StoreState = SnackbarSlice

export const useAppStore = create<StoreState>(
  devtools(
    (set, get, api) => ({
      ...snackbarSlice(set, get, api),
    }),
    { name: 'tce-store' }
  ) as () => StoreState
)
