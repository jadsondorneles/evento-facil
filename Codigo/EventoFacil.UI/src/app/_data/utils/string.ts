export const ehString = (input: unknown): input is string => {
  return typeof input === 'string'
}
