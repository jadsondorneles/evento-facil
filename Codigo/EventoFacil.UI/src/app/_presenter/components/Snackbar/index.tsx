'use client'

import React from 'react'

import { Box, Snackbar as MUISnackbar } from '@mui/material'
import { SnackbarCloseReason } from '@mui/material/Snackbar/Snackbar'

import { useAppStore } from '@/app/_data/store/store'

import Alert, { AlertPosition } from '../Alert'

const DEFAULT_AUTO_HIDE_DURATION_MS = 10000

const Snackbar: React.FC = () => {
  const { snackbar, hideSnackbar } = useAppStore()

  if (!snackbar) {
    return null
  }

  const autoHideDuration =
    snackbar.autoHideDuration === null ? null : snackbar.autoHideDuration || DEFAULT_AUTO_HIDE_DURATION_MS

  return (
    <MUISnackbar
      anchorOrigin={{
        vertical: snackbar.vertical ? snackbar.vertical : AlertPosition.top,
        horizontal: snackbar.horizontal ? snackbar.horizontal : AlertPosition.right,
      }}
      open={!!snackbar}
      onClose={(event: React.SyntheticEvent | Event, reason: SnackbarCloseReason) => {
        if (reason === 'timeout') {
          hideSnackbar()
        }
      }}
      autoHideDuration={autoHideDuration}
    >
      <Box>
        <Alert
          description={snackbar.description}
          severity={snackbar.severity}
          onClose={hideSnackbar}
          showIcon={snackbar.showIcon}
          html={snackbar.html}
        />
      </Box>
    </MUISnackbar>
  )
}

export default Snackbar
