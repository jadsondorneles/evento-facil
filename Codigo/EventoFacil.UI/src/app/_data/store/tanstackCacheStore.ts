import { QueryKey } from '@tanstack/query-core'
import { PersistedClient } from '@tanstack/react-query-persist-client'
import isEqual from 'lodash/isEqual'
import { create } from 'zustand'
import { persist } from 'zustand/middleware'

import { isClient } from '@/app/_data/utils/browser'
import ZustandStoreKeys from '@/app/_domain/constantes/zustandStoreKeys'

type TanstackCacheStoreType = {
  clear: () => void
  data: PersistedClient | null
  getQueryCache: <TData>(queryKey: QueryKey) => TData | null
  setData: (value: PersistedClient) => void
}

export const useTanstackCacheStore = create<TanstackCacheStoreType>()(
  persist(
    (set, get) => ({
      data: null,
      setData: (value: PersistedClient) => set({ data: value }),
      clear: () => set({ data: null }),
      getQueryCache: <TData>(queryKey: QueryKey) => {
        const query = get().data?.clientState.queries?.find((query) => isEqual(query.queryKey, queryKey))
        return query ? (query.state.data as TData) : null
      },
    }),
    {
      name: ZustandStoreKeys.tanstackCacheStore,
      skipHydration: !isClient(),
    }
  )
)
