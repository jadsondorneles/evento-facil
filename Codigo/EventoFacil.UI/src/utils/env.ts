import { PUBLIC_ENV_KEY } from 'next-runtime-env'

import { isClient } from '@/app/_data/utils/browser'

/**
 * Lê uma variável de ambiente segura do navegador ou de qualquer ambiente
 * variável do servidor (process.env) definida no arquivo @see {@link ./app/layout.tsx}.
 *
 * Uso:
 * ```ts
 * const API_URL = env('API_URL')
 * ```
 */
export function env(key: string): string | undefined {
  if (isClient()) {
    return window[PUBLIC_ENV_KEY]?.[key] || ''
  }
  return process.env[key]
}
