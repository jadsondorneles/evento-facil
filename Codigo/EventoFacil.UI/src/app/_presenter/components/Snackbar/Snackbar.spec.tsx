import React from 'react'

import { screen } from '@testing-library/react'

import { useAppStore } from '@/app/_data/store/store'

import { renderizarComTema } from '@/jest.utils'

import Snackbar from './index'

jest.mock('@/app/_data/store/store', () => ({
  useAppStore: jest.fn(),
}))

describe('Snackbar', () => {
  beforeEach(() => {
    ;(useAppStore as unknown as jest.Mock).mockReturnValue({
      snackbar: null,
      hideAlert: jest.fn(),
    })
  })

  it('deve corresponder ao snapshot', () => {
    ;(useAppStore as unknown as jest.Mock).mockReturnValue({
      snackbar: {
        description: 'Test Alert',
        severity: 'info',
      },
      hideAlert: jest.fn(),
    })
    const snackbarElement = renderizarComTema(<Snackbar />)
    expect(snackbarElement).toMatchSnapshot()
  })

  it('deve renderizar sem travar', () => {
    renderizarComTema(<Snackbar />)
  })

  it('não deve exibir nada se não houver alerta', () => {
    renderizarComTema(<Snackbar />)
    expect(screen.queryByRole('alert')).toBeNull()
  })

  it('deve exibir um alerta se existir no estado', () => {
    ;(useAppStore as unknown as jest.Mock).mockReturnValue({
      snackbar: {
        description: 'Test Alert',
        severity: 'info',
      },
      hideAlert: jest.fn(),
    })

    const snackbarElement = renderizarComTema(<Snackbar />)
    expect(snackbarElement.asFragment().querySelector(`.MuiSnackbar-root`)).not.toBeNull()
  })

  it('deve renderizar corretamente o conteúdo do suporte html no componente Alert', () => {
    const testHtmlContent = '<span>Teste o conteúdo HTML no Snackbar</span>'
    const snackbarState = {
      description: 'Test Snackbar',
      html: testHtmlContent,
    }

    ;(useAppStore as unknown as jest.Mock).mockReturnValue({
      snackbar: snackbarState,
      hideAlert: jest.fn(),
    })

    const { container } = renderizarComTema(<Snackbar />)
    const renderedHtmlContent = container.querySelector('span')
    expect(renderedHtmlContent).not.toBeNull()
    expect(renderedHtmlContent?.textContent).toBe('Teste o conteúdo HTML no Snackbar')
  })

  it('deve renderizar corretamente a descrição como uma lista de strings no componente Alert', () => {
    const testDescription = ['Test Descrição', 'Test Descrição 2']
    const snackbarState = {
      description: testDescription,
    }

    ;(useAppStore as unknown as jest.Mock).mockReturnValue({
      snackbar: snackbarState,
      hideAlert: jest.fn(),
    })

    const { container } = renderizarComTema(<Snackbar />)
    const renderedDescription = container.querySelectorAll('p')
    expect(renderedDescription).toHaveLength(2)
    expect(renderedDescription[0].textContent).toBe('Test Descrição')
    expect(renderedDescription[1].textContent).toBe('Test Descrição 2')
  })
})
