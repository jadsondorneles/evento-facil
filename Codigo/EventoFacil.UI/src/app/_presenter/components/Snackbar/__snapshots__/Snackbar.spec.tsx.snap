// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar deve corresponder ao snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiSnackbar-root MuiSnackbar-anchorOriginTopRight css-1arijsr-MuiSnackbar-root"
        role="presentation"
      >
        <div
          class="MuiBox-root css-0"
          direction="down"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-outlinedInfo MuiAlert-outlined css-1y4amgq-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-10jtlzh-MuiTypography-root"
              >
                Test Alert
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginTopRight css-1arijsr-MuiSnackbar-root"
      role="presentation"
    >
      <div
        class="MuiBox-root css-0"
        direction="down"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-outlinedInfo MuiAlert-outlined css-1y4amgq-MuiPaper-root-MuiAlert-root"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-10jtlzh-MuiTypography-root"
            >
              Test Alert
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
